File: Codeforces/p02a_Winner/winner.cpp
#include<bits/stdc++.h>

using namespace std;

int main(){
    int num;
    cin >> num;

    unordered_map<string, int> map_info;
    vector<string> playerName;
    vector<int> playerPoint;

    // Store all entries and calculate the final score
    for(int i = 0; i < num; i++){
        string name;
        int points;

        cin >> name >> points;

        playerName.push_back(name);
        playerPoint.push_back(points);

        // to check if the player's point record already exist or not
        if(map_info.find(name) != map_info.end()){
            map_info[name] += points;
        }else{
            map_info[name] = points;
        }

        // find the maximum score
        int maxFinalScore = INT_MIN;

        for(auto &entry : map_info){
            if(entry.second > maxFinalScore){
                maxFinalScore = entry.second;
            }
        }

        unordered_map<string, int> live_score;

        for(int i = 0; i < num; i++){
            string name = playerName[i];
            int point = playerPoint[i];

            if(live_score.find(name) != live_score.end()){
                live_score[name] += point;
            }else{
                live_score[name] = point;
            }

            int finalTotal = map_info[name];
            int currentTotal = live_score[name];

            if(finalTotal == maxFinalScore && currentTotal >= maxFinalScore){
                cout << name << endl;
                break;
            }
        }
    }

    return 0;
}

File: Codeforces/p02a_Winner/winner.py
num = int(input())

map_info = {}
playerName = []
playerPoint = []

for _ in range(num):
    name, points = input().split()
    points = int(points)

    playerName.append(name)
    playerPoint.append(points)

    # Check if the player's point record already exists
    if name in map_info:
        map_info[name] += points
    else:
        map_info[name] = points

    # Find the maximum Score
    maxFinalScore = float('-inf')

    for name2, score in map_info.items():
        if score > maxFinalScore:
            maxFinalScore = score

    liveScore = {}

    for i in range(num):
        name2 = playerName[i]
        point = playerPoint[i]

        if name2 in liveScore:
            liveScore[name] += points
        else:
            liveScore[name] = points

        finalTotal = map_info[name2]
        currentTotal = liveScore[name2]

        if finalTotal == maxFinalScore and currentTotal >= maxFinalScore:
            print(name2)
            break